### PR TITLE
Add Pallet2 and fn doctor_verifies

### PR DESCRIPTION
**Pull request by OverOrion**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -3,6 +3,7 @@ members = [
     "node",
     "pallets/template",
     "pallets/medical-record",
+    "pallets/medical-encryption/",
```
Comment (PieWol) [inline Cargo.toml:6]: Are we sure this is the right coupling "direction"?

File: pallets/medical-encryption/src/lib.rs [hunk 2]
```diff
@@ -25,7 +25,7 @@ pub mod pallet {
 
 	/// Configure the pallet by specifying the parameters and types on which it depends.
 	#[pallet::config]
-	pub trait Config: frame_system::Config {
+	pub trait Config: frame_system::Config + pallet_medical_record::Config {
```
Comment (PieWol) [inline pallets/medical-encryption/src/lib.rs]: Are we sure this is the right coupling "direction"?

**Review by PieWol — APPROVED** [review]
lgtm

File: pallets/medical-record/src/tests.rs [hunk 3]
```diff
@@ -100,6 +102,44 @@ fn doctor_can_add_record_for_patient() {
 	});
 }
 
+#[test]
+fn doctor_can_transform_unverified_record() {
```
Comment (LRUki) [inline pallets/medical-record/src/tests.rs:106]: Did you run the test? 
Seems like it's failing on my computer

File: pallets/medical-record/src/tests.rs [hunk 4]
```diff
@@ -100,6 +102,44 @@ fn doctor_can_add_record_for_patient() {
...
+		assert_ok!(MedicalRecord::create_account(doctor.clone(), UserType::Doctor));
+		assert_ok!(MedicalRecord::create_account(patient.clone(), UserType::Patient));
+
+		// Patient submits a record
+		assert_ok!(MedicalRecord::patient_adds_record(
+			patient.clone(),
+			BoundedVec::with_max_capacity()
+		));
+		// Doctor verifies it
+		let signature: BoundedVec<u8, ConstU32<3>> = bounded_vec![];
```
Comment (LRUki) [inline pallets/medical-record/src/tests.rs:119]: Could you use `MockSignatureLength` instead here?

Comment (LRUki) [reply pallets/medical-record/src/tests.rs:119]: oh it's merged. I will fix it in my branch dw

**Pull request merged**